python_modules/dagster/dagster/cli/dynamic_loader.py

```diff
@@ -208,68 +247,87 @@ def create_pipeline_loading_mode_data(info):
         raise InvalidPipelineLoadingComboError()
 
 
-DynamicObject = namedtuple('DynamicObject', 'module fn module_name fn_name object')
+class DynamicObject:
+    def __init__(self, module, module_name, fn_name):
+        self.module = module
+        self.module_name = module_name
+        self.fn_name = fn_name
+        self.after_load_fn = None
+        self.object = None
+        self.loaded = False
+
+    def load(self):
+        if self.loaded:
+            reloader.reload(self.module)
+        self.loaded = True
+
+        fn = getattr(self.module, self.fn_name)
+        check.is_callable(fn)
+        self.object = fn()
+        if self.after_load_fn:
+            self.object = self.after_load_fn(self.object)
```

Comment:
I'd prefer at this point to eliminate after_load_fn and move the type coercion back into this class as that is more clear. I think this generalization/abstraction actually obscures the intent. Make an issue to track this code cleanup issue.

Reply:
Cool. Reverted this change (and kept the exception we added if the type is not a Pipeline or Repository) and created #246.